buffer/src/main/java/io/netty/buffer/api/CompositeBuffer.java

```diff
@@ -791,6 +803,21 @@ public CompositeBuffer splitComponentsCeil(int splitOffset) {
         return buildSplitBuffer(splits);
     }
 
+    /**
+     * Break a composite buffer into its constituent components.
+     * <p>
+     * This "consumes" the composite buffer, leaving the composite buffer instance as if it had been closed.
+     * The buffers in the returned array are not closed, and become owned by the caller.
+     *
+     * @return An array of the constituent buffer components.
+     */
+    public Buffer[] decomposeBuffer() {
+        Buffer[] result = bufs;
+        bufs = EMPTY_BUFFER_ARRAY;
```

Comment:
should we only do this after we call close as `close()` might throw ?

Reply:
We need to grab the array before calling close as otherwise the component buffers will be closed as well, but there is an interesting point about close throwing, which needs to be handled.

Reply:
Can we add a test-case for this ?